test_collections/sdk_tests/support/python_testing/models/python_test_parser.py

```diff
@@ -73,23 +73,31 @@ def __parse_test_case_from_class(
     pics_method_name = "pics_" + tc_name
 
     methods = [m for m in class_.body if isinstance(m, ast.FunctionDef)]
+
+    tc_desc = tc_name
+    tc_steps = []
+    tc_pics = []
+
     try:
         desc_method = next(m for m in methods if desc_method_name in m.name)
         tc_desc = desc_method.body[BODY_INDEX].value.value  # type: ignore
+    except StopIteration:  # Raised when `next` doesn't find a matching method
```

Comment:
We can also just pass a default value of `None` to `next`.

eg. 

```py
desc_method = next(m for m in methods if desc_method_name in m.name, None)
if desc_method:
    tc_desc = desc_method.body[BODY_INDEX].value.value
```

Reply:
have we considered putting this in a helper?
 
 ```py
 def get_method_by_name(name:str, methods:list) -> Optional[method]:
     return next(m for m in methods if name in m.name, None)
 ```
 
 Not sure if it's helpful